Unable to delete packages installed as editable from Windows junction link

### Description

If a package with a path containing a Windows junction link is installed as editable, an egg-link `AssertionError` occurs that the package path does not match when uninstalling.

```
AssertionError: Egg-link \path\that\link\include\test_pkg does not match installed location of test-pkg (at \real\package\path)
```


### Expected behavior

Deleting or reinstalling packages should not raise `AssertionError: Egg-link`.

### pip version

21.2.2

### Python version

3.8.8

### OS

Windows 10 (version 1909)

### How to Reproduce

```batch
# Create or locate test package to original directory
code src\test_pkg

# Make junction link
mklink /J src-link src

# Install package over junction link path
cd src-link\test_pkg
pip install -e .

# Uninstall or re-install package over junction link
pip uninstall test_pkg

>> Raise AssertionError
```

### Output

```sh-session
ERROR: Exception:
Traceback (most recent call last):
  File "C:\Users\jiunbae\conda\lib\site-packages\pip\_internal\cli\base_command.py", line 173, in _main
    status = self.run(options, args)
  File "C:\Users\jiunbae\conda\lib\site-packages\pip\_internal\cli\req_command.py", line 203, in wrapper
    return func(self, options, args)
  File "C:\Users\jiunbae\conda\lib\site-packages\pip\_internal\commands\install.py", line 390, in run
    installed = install_given_reqs(
  File "C:\Users\jiunbae\conda\lib\site-packages\pip\_internal\req\__init__.py", line 66, in install_given_reqs
    uninstalled_pathset = requirement.uninstall(
  File "C:\Users\jiunbae\conda\lib\site-packages\pip\_internal\req\req_install.py", line 632, in uninstall
    uninstalled_pathset = UninstallPathSet.from_dist(dist)
  File "C:\Users\jiunbae\conda\lib\site-packages\pip\_internal\req\req_uninstall.py", line 527, in from_dist
    assert (link_pointer == dist.location), (
AssertionError: Egg-link e:\workspace-link\test_pkg does not match installed location of test-pkg (at e:\workspace\test_pkg)
```


### Code of Conduct

- [X] I agree to follow the [PSF Code of Conduct](https://www.python.org/psf/conduct/).

--- a/news/10696.bugfix.rst
+++ b/news/10696.bugfix.rst
@@ -1,0 +1,1 @@
+Fix uninstall editable from Windows junction link

--- a/src/pip/_internal/req/req_uninstall.py
+++ b/src/pip/_internal/req/req_uninstall.py
@@ -526,10 +526,8 @@ class UninstallPathSet:
             # PEP 660 modern editable is handled in the ``.dist-info`` case
             # above, so this only covers the setuptools-style editable.
             with open(develop_egg_link) as fh:
-                link_pointer = os.path.normcase(
-                    os.path.realpath(fh.readline().strip())
-                )
-            assert link_pointer == dist_location, (
+                link_pointer = os.path.normcase(fh.readline().strip())
+            assert os.path.samefile(link_pointer, dist_location), (
                 f"Egg-link {link_pointer} does not match installed location of "
                 f"{dist.raw_name} (at {dist_location})"
             )

--- a/src/pip/_internal/req/req_uninstall.py
+++ b/src/pip/_internal/req/req_uninstall.py
@@ -526,7 +526,9 @@ class UninstallPathSet:
             # PEP 660 modern editable is handled in the ``.dist-info`` case
             # above, so this only covers the setuptools-style editable.
             with open(develop_egg_link) as fh:
-                link_pointer = os.path.normcase(fh.readline().strip())
+                link_pointer = os.path.normcase(
+                    os.path.realpath(fh.readline().strip())
+                )
             assert link_pointer == dist_location, (
                 f"Egg-link {link_pointer} does not match installed location of "
                 f"{dist.raw_name} (at {dist_location})"

